src/dmd/traits.d

```diff
@@ -924,12 +924,34 @@ extern (C++) Expression semanticTraits(TraitsExp e, Scope* sc)
                 else
                     ex = dve.e1;
             }
+            else if (ex.op == TOK.template_)
+            {
+                VarExp ve = cast(VarExp)ex;
+                auto td = ve.var.isTemplateDeclaration();
+                f = td;
```

Comment:
`f = ve.var.isTeplateDeclaration` and you can get rid of `td`

Reply:
No can do - I use `td.funcroot` later, and that's defined in `TemplateDeclaration`, not in `Dsymbol`.